src/httpclient.cc

```diff
@@ -16,6 +18,12 @@
 #include "logger.h"
 #include "openssl_compat.h"
 
+#ifdef TEST_PKCS11_ENGINE_PATH
```

Comment:
I would like to encapsulate all pkcs11 functionality to separate class, currently I created p11 class, but maybe it is slightly  wrong name.

Reply:
Maybe "crypto" is more appropriate place for everything related to OpenSSL?